packages/stems/src/components/HarmonyButton/HarmonyButton.tsx

```diff
@@ -0,0 +1,105 @@
+import { CSSProperties, forwardRef } from 'react'
+
+import cn from 'classnames'
+
+import { useMediaQueryListener } from 'hooks/useMediaQueryListener'
+import { CSSCustomProperties } from 'styles/types'
+import { toCSSVariableName } from 'utils/styles'
+
+import styles from './HarmonyButton.module.css'
+import {
+  HarmonyButtonProps,
+  HarmonyButtonType,
+  HarmonyButtonSize
+} from './types'
+
+const SIZE_STYLE_MAP: { [k in HarmonyButtonSize]: string } = {
+  [HarmonyButtonSize.SMALL]: styles.small,
+  [HarmonyButtonSize.DEFAULT]: styles.default,
+  [HarmonyButtonSize.LARGE]: styles.large
+}
+
+const TYPE_STYLE_MAP: { [k in HarmonyButtonType]: string } = {
+  [HarmonyButtonType.PRIMARY]: styles.primary,
+  [HarmonyButtonType.SECONDARY]: styles.secondary,
+  [HarmonyButtonType.TERTIARY]: styles.tertiary,
+  [HarmonyButtonType.DESTRUCTIVE]: styles.destructive,
+  [HarmonyButtonType.GHOST]: styles.ghost
+}
+
+/**
+ * A common Button component. Includes a few variants and options to
+ * include and position icons.
+ */
+export const HarmonyButton = forwardRef<HTMLButtonElement, HarmonyButtonProps>(
+  function Button(
+    {
+      color,
+      text,
+      variant = HarmonyButtonType.PRIMARY,
+      size = HarmonyButtonSize.DEFAULT,
+      leftIcon: LeftIconComponent,
+      rightIcon: RightIconComponent,
+      disabled,
+      widthToHideText,
+      minWidth,
+      className,
+      iconClassName,
```

Comment:
so as a pattern going forward we want nested classnames to be indivudally named? in mobile and other libraries ive seen the `classes` pattern where it's an object of all the inner classnames we would want to open up.

Reply:
I will say that with the `classes` pattern, it might be harder to write types such that you know what the valid class names are.
For example,
```
// Harder to write a props shape that will catch this typo!
<HarmonyButton classes={icn: styles.icon, text: styles.text} />
```

Not impossible, I could whip up a generic that makes it easier to define. Just wanted to point that out.

Reply:
yah totally agreed not having to pass styles in, so just removing it is great for now, and then we can all have a discussion about what pattern we want moving forward!! as far as classes type utility we added something somewhat basic in mobile that works well, so can prob reuse that if we go forward with classes